### PR TITLE
Features/#398 remove slashes in paths

### LINKED ISSUE #398: Paths contain hard-coded slashes in some tasks
Old topic, but still pending:
The following tasks make use of absolute file paths which (beside of coding standards) most likely prevents the pipeline to be run under Windows (even if users of this so-called OS is not our primary target group :penguin:):
- [x] demandregio @ClaraBuettner @IlkaCu 
- [x] geothermal @ClaraBuettner 
- [x] scenario_capacities @ClaraBuettner 
- [x] calculate_dlr @ClaraBuettner 
- [x] insert-gas-grid @AmeliaNadal 

(list might be incomplete)
Low prio but it's easy to fix and saves some pain for other users.

See @gnn's post on this topic: https://github.com/openego/eGon-data/pull/338#discussion_r673135401
Related: #59

### PR DESCRIPTION
**Pull request by AmeliaNadal**
The paths with slashes have been replaced with Path objects.
Only tested in test mode (but it shouldn't make a difference).

Close #398

### CODE REVIEW

File: src/egon/data/processing/calculate_dlr.py [hunk 1]
```diff
@@ -24,10 +24,14 @@ def Calculate_DLR():
 
     """
 
-    weather_info_path = "cutouts/europe-2011-era5/201101.nc"
+    weather_info_path = (
+        Path(".") / "cutouts" / "germany-2011-era5.nc")
+
```
Comment (nesnoj) [inline src/egon/data/processing/calculate_dlr.py:29]: @AmeliaNadal The reason why u changed the filename here is not related to the actual purpose of this PR, is it?

**Review by nesnoj — COMMENTED** [review]
@AmeliaNadal Have you tested your changes in SH mode?

File: src/egon/data/processing/calculate_dlr.py (hunk 1 above)
Comment (AmeliaNadal) [reply src/egon/data/processing/calculate_dlr.py:29]: @nesnoj This file name change is actually part of Clara's commit so I cannot tell more about it, sorry.

I had tested the my changes in SH mode. They were successful so haven't paid enough attention to the other tasks before making the PR (sorry!). However, I rerun the worflow and the era5.download-era5 task is falling. I am getting the following error:
```
*** Reading local file: /opt/egon_data_workflow/wd_test_mode/airflow/logs/egon-data-processing-pipeline/era5.download-era5/2021-08-24T09:41:55.939762+00:00/1.log
[2021-08-24 11:51:30,303] {taskinstance.py:670} INFO - Dependencies all met for <TaskInstance: egon-data-processing-pipeline.era5.download-era5 2021-08-24T09:41:55.939762+00:00 [queued]>
[2021-08-24 11:51:30,463] {taskinstance.py:670} INFO - Dependencies all met for <TaskInstance: egon-data-processing-pipeline.era5.download-era5 2021-08-24T09:41:55.939762+00:00 [queued]>
[2021-08-24 11:51:30,464] {taskinstance.py:880} INFO - 
--------------------------------------------------------------------------------
[2021-08-24 11:51:30,464] {taskinstance.py:881} INFO - Starting attempt 1 of 1
[2021-08-24 11:51:30,464] {taskinstance.py:882} INFO - 
--------------------------------------------------------------------------------
[2021-08-24 11:51:38,107] {taskinstance.py:901} INFO - Executing <Task(Era5 (versioned)): era5.download-era5> on 2021-08-24T09:41:55.939762+00:00
[2021-08-24 11:51:38,115] {standard_task_runner.py:54} INFO - Started process 18380 to run task
[2021-08-24 11:51:38,186] {standard_task_runner.py:77} INFO - Running: ['airflow', 'run', 'egon-data-processing-pipeline', 'era5.download-era5', '2021-08-24T09:41:55.939762+00:00', '--job_id', '2976', '--pool', 'default_pool', '--raw', '-sd', 'DAGS_FOLDER/dags/pipeline.py', '--cfg_path', '/tmp/tmp62agmu9v']
[2021-08-24 11:51:38,188] {standard_task_runner.py:78} INFO - Job 2976: Subtask era5.download-era5
[2021-08-24 11:51:47,312] {logging_mixin.py:120} INFO - Running <TaskInstance: egon-data-processing-pipeline.era5.download-era5 2021-08-24T09:41:55.939762+00:00 [running]> on host torch.esy.ve.dlr
[2021-08-24 11:52:13,228] {taskinstance.py:1150} ERROR - join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
Traceback (most recent call last):
  File "/home/nada_am/anaconda3/envs/egondata_l/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/opt/egon_data_workflow/egondata_test_mode/eGon-data/src/egon/data/datasets/__init__.py", line 192, in skip_task
    result = super(type(task), task).execute(*xs, **ks)
  File "/home/nada_am/anaconda3/envs/egondata_l/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 113, in execute
    return_value = self.execute_callable()
  File "/home/nada_am/anaconda3/envs/egondata_l/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/egon_data_workflow/egondata_test_mode/eGon-data/src/egon/data/datasets/era5.py", line 115, in download_era5
    cutout = import_cutout()
  File "/opt/egon_data_workflow/egondata_test_mode/eGon-data/src/egon/data/datasets/era5.py", line 88, in import_cutout
    cutout = atlite.Cutout(
  File "/home/nada_am/anaconda3/envs/egondata_l/lib/python3.8/site-packages/atlite/cutout.py", line 46, in __init__
    self.cutout_dir = os.path.join(cutout_dir, name)
  File "/home/nada_am/anaconda3/envs/egondata_l/lib/python3.8/posixpath.py", line 90, in join
    genericpath._check_arg_types('join', a, *p)
  File "/home/nada_am/anaconda3/envs/egondata_l/lib/python3.8/genericpath.py", line 152, in _check_arg_types
    raise TypeError(f'{funcname}() argument must be str, bytes, or '
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
[2021-08-24 11:52:13,414] {taskinstance.py:1187} INFO - Marking task as FAILED. dag_id=egon-data-processing-pipeline, task_id=era5.download-era5, execution_date=20210824T094155, start_date=20210824T095130, end_date=20210824T095213
[2021-08-24 11:52:49,923] {local_task_job.py:102} INFO - Task exited with return code 1
```

**Comment by nesnoj** [comment]
DE run was successful too, I'll merge...

**Pull request merged**